service/plugin/src/main/java/com/solace/maas/ep/event/management/agent/plugin/route/handler/base/AbstractRouteBuilder.java

```diff
@@ -40,7 +41,8 @@ protected AbstractRouteBuilder(ScanTypeDescendentsProcessor scanTypeDescendentsP
     public void configure() throws Exception {
 
         // NetworkClient Selectable network i/o exception + Kafka Future exceptions
-        onException(IOException.class, InterruptedException.class, ExecutionException.class, TimeoutException.class)
+        onException(IOException.class, InterruptedException.class, ExecutionException.class, TimeoutException.class,
+                PluginClientException.class)
```

Comment:
Adding the `PluginClientException` that any plugin can throw. This signals to camel that an error has occurred and Camel can set the scan type status to `FAILED`.